crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -29,7 +29,7 @@ pub struct DynamicScene {
 /// A reflection-powered serializable representation of an entity and its components.
 pub struct DynamicEntity {
     /// The transiently unique identifier of a corresponding `Entity`.
-    pub entity: u32,
+    pub entity: u64,
```

Comment:
At this point, why not store a proper `Entity` here?

Reply:
A benefit to doing it this way is that it's backwards compatible with existing scenes in most scene formats. They'll deserialize as <whatever>v0 as they do today, and as a side effect using an `Entity` here will increase the blast radius somewhat (breaks more existing tests and, if I recall correctly, examples). I also kinda prefer the serialization from `to_bits` instead of the struct serialization you get from putting an `Entity` here as it's more succinct and generation/id should be internal anyway.

Reply:
Okay, I'm happy with that justification.

Reply:
I would use `Entity` for clarity.

Reply:
In that case would it maybe be reasonable to also change `Entity` to (de)serialize as `to_bits` and `from_bits`?

Reply:
What benefits does it provide?

Reply:
In this context, the aforementioned backwards compatibility with existing scenes is a direct technical benefit. But in general, it keeps generation/index a little more internal, since `Entity` should be treated as an opaque identifier by users and, probably, by anything outside `bevy_ecs`, at least in my opinion. Within bevy itself, the only place generation or index is used outside `bevy_ecs` at the moment is a tiny bit of `bevy_sprite`. It'd actually be quite possible to move the accessors to `pub(crate)`.

Reply:
Oh actually, I suppose if we switch to `Entity` and change the serialization it breaks backwards compatibility anyway because it'd change how component entity references serialize. So the only thing that doesn't break compatibility is the current `u64` approach.

Reply:
I would break the compatibility. We did it literally in the previous release. But it will result in a cleaner API.

Reply:
Okay I went with that. I still think changing `Entity`'s serialization is probably worth doing but it doesn't need to be coupled with this change.